components/apimgt/org.wso2.carbon.apimgt.persistence.mongodb/src/main/java/org/wso2/carbon/apimgt/persistence/mongodb/MongoDBPersistenceImpl.java

```diff
@@ -0,0 +1,813 @@
+        MongoDBDevPortalAPI mongoDBAPIDocument = collection.find(eq("_id", new ObjectId(apiId))).first();
+        if (mongoDBAPIDocument == null) {
+            String msg = "Failed to get API. " + apiId + " does not exist in mongodb database";
+            log.error(msg);
+            throw new APIPersistenceException(msg);
+        }
+        return MongoAPIMapper.INSTANCE.toDevPortalApi(mongoDBAPIDocument);
+    }
+
+    @Override
+    public void deleteAPI(Organization org, String apiId) throws APIPersistenceException {
+        MongoCollection<MongoDBPublisherAPI> collection = getPublisherCollection(org.getName());
+        collection.deleteOne(eq("_id", new ObjectId(apiId)));
+        log.info("successfully deleted " + apiId + " from mongodb");
+    }
+
+    @Override
+    public PublisherAPISearchResult searchAPIsForPublisher(Organization org, String searchQuery, int start,
+                                                           int offset, UserContext ctx) throws APIPersistenceException {
+        searchQuery = "";
+        int skip = start;
+        int limit = offset;
+        MongoCollection<MongoDBPublisherAPI> collection = getPublisherCollection(ctx.getOrganization().getName());
+        long totalCount = collection.countDocuments();
+        MongoCursor<MongoDBPublisherAPI> aggregate = collection.aggregate(getPublisherSearchAggregate(searchQuery, skip, limit))
+                .cursor();
+        PublisherAPISearchResult publisherAPISearchResult = new PublisherAPISearchResult();
+        List<PublisherAPIInfo> publisherAPIInfoList = new ArrayList<>();
+
+        while (aggregate.hasNext()) {
+            MongoDBPublisherAPI mongoDBAPIDocument = aggregate.next();
+            PublisherAPIInfo api = MongoAPIMapper.INSTANCE.toPublisherApi(mongoDBAPIDocument);
+            publisherAPIInfoList.add(api);
+        }
+        publisherAPISearchResult.setPublisherAPIInfoList(publisherAPIInfoList);
+        publisherAPISearchResult.setReturnedAPIsCount(publisherAPIInfoList.size());
+        publisherAPISearchResult.setTotalAPIsCount((int) totalCount);
+        return publisherAPISearchResult;
+    }
+
+    private String getSearchQuery(String query) {
+
+        if (!query.contains(":")) {
+            return "*" + query + "*";
+        }
+        String[] queryArray = query.split(":");
+        String searchCriteria = queryArray[0];
+        String searchQuery = queryArray[1];
+        return "*" + searchQuery + "*";
+    }
+
+    private List<Document> getPublisherSearchAggregate(String query, int skip, int limit) {
+        String searchQuery = getSearchQuery(query);
+        List<String> paths = new ArrayList<>();
+        paths.add("apiName");
+        paths.add("providerName");
+        paths.add("version");
+        paths.add("context");
+//        paths.add("status");
+//        paths.add("description");
+//        paths.add("tags");
+//        paths.add("gatewayLabels");
+//        paths.add("additionalProperties");
```

Comment:
can we remove this variable if it is not used?

Reply:
its not used right now but will be needed when implementing the content search part, therefore i left it commented, is that ok?